target/ppc/mmu_common.c

```diff
@@ -618,14 +618,15 @@ static int mmubooke_check_tlb(CPUPPCState *env, ppcemb_tlb_t *tlb,
 
 found_tlb:
 
-    if (msr_pr != 0) {
+    if (env->msr & M_MSR_PR) {
         prot2 = tlb->prot & 0xF;
     } else {
         prot2 = (tlb->prot >> 4) & 0xF;
     }
 
     /* Check the address space */
-    if ((access_type == MMU_INST_FETCH ? msr_ir : msr_dr) != (tlb->attr & 1)) {
+    if ((access_type == MMU_INST_FETCH ?
+        (env->msr & M_MSR_IR) : (env->msr & M_MSR_DR)) != (tlb->attr & 1)) {
```

Comment:
I think this one is incorrect, we have `(env->msr & (1ull << MSR_IR)) != (tlb->attr & 1)` which is always true, or the same case with `MSR_DR` instead of `MSR_IR` which is also always true